packages/block-editor/src/components/inserter/index.js

```diff
@@ -105,10 +119,67 @@ class Inserter extends Component {
 
 export default compose( [
 	withSelect( ( select, { rootClientId } ) => {
-		const { hasInserterItems } = select( 'core/block-editor' );
-
+		const {
+			hasInserterItems,
+			__experimentalHasOnlyOneAllowedBlockType,
+			__experimentalGetTheOnlyAllowedBlockType,
+		} = select( 'core/block-editor' );
+		const allowedBlock = getBlockType( __experimentalGetTheOnlyAllowedBlockType( rootClientId ) );
 		return {
 			hasItems: hasInserterItems( rootClientId ),
+			hasOnlyOneAllowedInserterItem: __experimentalHasOnlyOneAllowedBlockType( rootClientId ),
+			blockTitle: allowedBlock ? allowedBlock.title : '',
+		};
+	} ),
+	withDispatch( ( dispatch, ownProps, { select } ) => {
+		return {
+			insertTheOnlyAllowedItem: () => {
+				const { rootClientId, clientId, isAppender, destinationRootClientId } = ownProps;
+				const {
+					__experimentalHasOnlyOneAllowedBlockType,
+					__experimentalGetTheOnlyAllowedBlockType,
+				} = select( 'core/block-editor' );
+
+				const hasOnlyOneAllowedInserterItem = __experimentalHasOnlyOneAllowedBlockType( rootClientId );
+
+				if ( ! hasOnlyOneAllowedInserterItem ) {
+					return false;
+				}
+
+				const parentAllowedBlocks = __experimentalGetTheOnlyAllowedBlockType( rootClientId );
+
+				function getInsertionIndex() {
```

Comment:
I see this is also defined in the inserter menu and now used in three places. I wonder if it can be consolidated. I was thinking of whether a new version of `insertBlock` could be made that has this logic built-in.

Maybe one for a separate PR after this is merged?

Reply:
yes, i tried making this a selector but for some reason it didn't work "out of the box", as the position was not the expected one (probably easy to fix if you know what's wrong, but in my case easier to revert).

I would do it in a separate PR and do it like you said above, it's even better :)